src/DateTimePicker/DateTime.elm

```diff
@@ -372,9 +372,9 @@ validate datetime =
     else if datetime.day < 1 then
         Nothing
     else if datetime.month == Time.Feb then
-        if isLeapYear datetime.year && datetime.day > 29 then
+        if not (isLeapYear datetime.year) && datetime.day > 28 then
```

Comment:
So this month, selections were failing this first `isLeapYear datetime.year && datetime.day > 29` because the day wasn't greater than `29`, and then skipping to the next clause where the day was indeed greater than `28`, causing the this function to return `Nothing`, and the input eventually being cleared.

Instead, we catch all of the non-leap-year cases first, and then go to the `> 29` leap-year specific check, and otherwise return a valid datetime.

I think the new logic looks right. 👍